### DISCUSSION TITLE
What if two "Shared Business Folders" have the same name?

### DISCUSSION THREAD

**Discussion by rrodrigueznt** [abraunegg/onedrive#2119] [General]
Hi! In a [comment ](https://github.com/abraunegg/onedrive/issues/1300#issuecomment-786432069)to Feature Request #1300 @abraunegg wrote:
> The only _real_ way to support syncing business shared individual files at present would be to re-write the entire feature so that they are correctly supported with pattern matching between file / folder designations. This would also require a complete re-write of how the folders are stored and represented locally. This would also open up a whole host of other questions & issues - if you have a list of files, and someone shares that file with you, with the same name as someone else, what happens - conflict? overwrite? fail? something else?

Currently it is also possible to face a situation where two, or more, shared folders have the same name. For instance, from my list of shared folders:
```
Shared Folder:   2021
Shared Folder:   2021
Shared Folder:   2022
Shared Folder:   20220302
Shared Folder:   20220428
Shared Folder:   20220509
```
Before going ahead and adding those folders to the `business_shared_folders` file, please, what should I expect in that situation? Special thanks to @abraunegg for his patience and great work!

**Comment by abraunegg** [comment]
@rrodrigueznt 
If two shared folders have the same name, the first share will be used. The second will not be used.

This is because, when the original feature was written there not was enough user testing of the feature before it was implemented.

When will this be fixed?
* The entire client needs a full re-write and will be considered as part of that
* This is flagged as part of https://github.com/abraunegg/onedrive/issues/1251
